go/libkb/keyfamily.go

```diff
@@ -1198,17 +1199,32 @@ func (ckf *ComputedKeyFamily) getDeviceForKidHelper(kid keybase1.KID) (ret *Devi
 	return
 }
 
-func (ckf *ComputedKeyFamily) GetAllDevices() []*Device {
-	devices := []*Device{}
+type byAge []*Device
+
+func (a byAge) Len() int           { return len(a) }
+func (a byAge) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a byAge) Less(i, j int) bool { return a[i].CTime < a[j].CTime }
+
+func (ckf *ComputedKeyFamily) GetAllDevices() []DeviceWithDeviceNumber {
+	devicesNoNum := make([]*Device, 0, len(ckf.cki.Devices))
 	for _, device := range ckf.cki.Devices {
-		devices = append(devices, device)
+		devicesNoNum = append(devicesNoNum, device)
+	}
+	sort.Sort(byAge(devicesNoNum))
+	devices := make([]DeviceWithDeviceNumber, 0, len(devicesNoNum))
+	deviceNumMap := make(map[string]int)
+	for _, device := range devicesNoNum {
+
+		devices = append(devices, DeviceWithDeviceNumber{device, deviceNumMap[device.Type]})
```

Comment:
the linter didn't complain here?  I think you need to specify the struct fields.

Reply:
It did not complain. Would you like me to specify them anyway?